src/tensorwaves/optimizer/callbacks.py

```diff
@@ -6,6 +6,7 @@
 
 import tensorflow as tf
 import yaml
+from pandas import DataFrame
```

Comment:
I would use `import pandas as pd` as to distinguish namespaces in the rest of the code below.